xla/service/gpu/fusions/transpose_mlir.cc

```diff
@@ -91,11 +136,20 @@ Tiling ComputeTransposeTiling(const TransposeDescription& tiled_transpose) {
                                              transposed_dims[permutation[1]],
                                              transposed_dims[permutation[2]]};
 
+  int vector_size =
+      GetVectorSize(analysis, input_dims, permutation, kNumRows * WarpSize());
+
   // We tile along the minor dimensions pre- and post-transpose.
   absl::InlinedVector<int64_t, 4> tile_sizes{1, 1, 1};
-  tile_sizes[permutation[2]] = WarpSize() / kNumRows;
+  tile_sizes[permutation[2]] = WarpSize() * vector_size / kNumRows;
```

Comment:
This doesn't look right. In the vectorized case you end up with:

```
tile_sizes {1, 16, 1, 2}
```

In the unvectorized case, you get

```
tile_sizes {1, 8, 1, 1}
```

So you actually do `vector_size^2` amount of work per thread.

Reply:
See the below example:
If threads = [1, 4, 1, 32] and tile_sizes = [1, 16, 1, 2], we can see 4 * 16 == 32 * 2. As you mentioned, pre-transpose threads is same as post-transpose threads. If we set tile_sizes as [1, 8, 1, 2], each warp write 8 * 4 elements to memory, which is not same as 2 * 32.